scenes/ui/story_quest/item_slot.gd

```diff
@@ -0,0 +1,43 @@
+# SPDX-FileCopyrightText: The Threadbare Authors
+# SPDX-License-Identifier: MPL-2.0
+extends TextureRect
+
+## UI slot for [InventoryItem]s.
+
+var filled_with_item: InventoryItem = null:
+	set(new_item):
+		filled_with_item = new_item
+		texture = filled_with_item.texture()
+
+
+## Shows the collected [InventoryItem] in this item slot without animation.
+func start_as_filled(inventory_item: InventoryItem) -> void:
+	if is_filled():
+		return
+
+	filled_with_item = inventory_item
+	modulate = Color.WHITE
+
+
+func is_filled() -> bool:
+	return filled_with_item != null
+
+
+## Shows the collected [InventoryItem] in this item slot with a quick animation.
+func fill(inventory_item: InventoryItem) -> void:
+	if is_filled():
+		return
+
+	filled_with_item = inventory_item
+	texture = inventory_item.texture()
+	create_tween().tween_property(self, "modulate", Color.WHITE, 0.4).set_trans(Tween.TRANS_ELASTIC)
+	pivot_offset = size / 2.0
+	create_tween().tween_property(self, "rotation", 0.0, 0.4).from(-PI).set_trans(Tween.TRANS_QUAD)
+	await (
+		create_tween()
+		. tween_property(self, "scale", Vector2.ONE * 2.0, 0.4)
+		. from(Vector2.ONE)
+		. set_trans(Tween.TRANS_QUAD)
+		. finished
+	)
+	create_tween().tween_property(self, "scale", Vector2.ONE, 0.5).set_trans(Tween.TRANS_QUAD)
```

Comment:
An alternative to making this animation programmatically is to add an AnimationPlayer node to the ItemSlot scene and call play() here. I wonder if that is more user friendly for learners coming to this project.

Reply:
agree, I think using animations for this will be better